homeassistant/components/switch/zigate.py

```diff
@@ -0,0 +1,124 @@
+"""
+ZiGate platform.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/ZiGate/
+"""
+import logging
+from homeassistant.components.switch import SwitchDevice
+
+DOMAIN = 'zigate'
+DATA_ZIGATE_DEVICES = 'zigate_devices'
+DATA_ZIGATE_ATTRS = 'zigate_attributes'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the ZiGate sensors."""
+    if discovery_info is None:
+        return
+
+    z = hass.data[DOMAIN]
+    import zigate
+
+    def sync_attributes():
```

Comment:
A similar function is used in all platforms. Try to extract it into a general function that can be moved to the component and then imported and used in the platforms.